### PR TITLE
Add support for CUDA 10.1

### PR DESCRIPTION
**Pull request by BenjaminW3**
Implements #758

### CODE REVIEW

File: alpakaConfig.cmake [hunk 1]
```diff
@@ -507,6 +507,10 @@ IF(ALPAKA_ACC_GPU_CUDA_ENABLE)
                         IF(CMAKE_CXX_COMPILER_VERSION GREATER_EQUAL 8.0)
                             MESSAGE(FATAL_ERROR "NVCC 10.0 does not support GCC 8+. Please use GCC 4.9, 5, 6 or 7!")
                         ENDIF()
+                    ELSEIF(CUDA_VERSION VERSION_EQUAL 10.1)
+                        IF(CMAKE_CXX_COMPILER_VERSION GREATER_EQUAL 9.0)
+                            MESSAGE(FATAL_ERROR "NVCC 10.0 does not support GCC 9+. Please use GCC 4.9, 5, 6, 7 or 8!")
```
Comment (sbastrakov) [inline alpakaConfig.cmake]: This should probably be `NVCC 10.1`?

File: alpakaConfig.cmake [hunk 2]
```diff
@@ -529,6 +533,10 @@ IF(ALPAKA_ACC_GPU_CUDA_ENABLE)
                         IF(CMAKE_CXX_COMPILER_VERSION GREATER_EQUAL 7.0)
                             MESSAGE(FATAL_ERROR "NVCC 10.0 does not support clang 7+. Please use clang 4, 5 or 6!")
                         ENDIF()
+                    ELSEIF(CUDA_VERSION VERSION_EQUAL 10.1)
+                        IF(CMAKE_CXX_COMPILER_VERSION GREATER_EQUAL 8.0)
+                            MESSAGE(FATAL_ERROR "NVCC 10.0 does not support clang 8+. Please use clang 4, 5, 6 or 7!")
```
Comment (sbastrakov) [inline alpakaConfig.cmake]: Same as my previous comment.

File: alpakaConfig.cmake (hunk 1 above)
Comment (BenjaminW3) [reply alpakaConfig.cmake]: fixed

**Comment by BenjaminW3** [comment]
Ready for approval and merging!

**Review by sbastrakov — APPROVED** [review]

**Pull request merged**